uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/test/java/org/uberfire/ext/metadata/io/BatchIndexKeepFileTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2020 JBoss, by Red Hat, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.uberfire.ext.metadata.io;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Executors;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.apache.lucene.search.TermQuery;
+import org.junit.Test;
+import org.uberfire.commons.async.DescriptiveThreadFactory;
+import org.uberfire.ext.metadata.engine.BatchIndexListener;
+import org.uberfire.ext.metadata.engine.Indexer;
+import org.uberfire.ext.metadata.engine.Observer;
+import org.uberfire.ext.metadata.io.lucene.BaseIndexTest;
+import org.uberfire.ext.metadata.model.KCluster;
+import org.uberfire.ext.metadata.model.KObject;
+import org.uberfire.ext.metadata.model.KObjectKey;
+import org.uberfire.io.attribute.DublinCoreView;
+import org.uberfire.java.nio.file.Path;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.uberfire.ext.metadata.io.KObjectUtil.toKCluster;
+
+public class BatchIndexKeepFileTest extends BaseIndexTest {
+
+    @Override
+    protected String[] getRepositoryNames() {
+        return new String[]{"temp-repo-test"};
+    }
+
+    @Test
+    public void testIndex() {
+
+        {
+            final Path file = ioService().get("git://temp-repo-test/xxx/.keep");
+            ioService().write(file,
+                              "");
+        }
```

Comment:
Using such block inside a test is quite new for me. Could you please explain?

Reply:
This block trend is popular in index related tests. I think I'll change to a method here, since this setup is a lot more simple than the other tests.